src/connector/src/source/kinesis/source/message.rs

```diff
@@ -14,14 +14,25 @@
 
 use aws_sdk_kinesis::types::Record;
 use aws_smithy_types_convert::date_time::DateTimeExt;
+use risingwave_common::types::{DatumRef, ScalarRefImpl};
 
 use crate::source::{SourceMessage, SourceMeta, SplitId};
 
 #[derive(Clone, Debug)]
 pub struct KinesisMeta {
-    // from `approximate_arrival_timestamp` of type `Option<aws_smithy_types::DateTime>`
-    #[expect(dead_code)]
-    timestamp: Option<i64>,
+    timestamp: Option<aws_smithy_types::DateTime>,
+}
+
+impl KinesisMeta {
+    pub fn extract_timestamp(&self) -> Option<DatumRef<'_>> {
+        self.timestamp
+            .and_then(|dt| dt.to_chrono_utc().ok().map(|dt| dt.timestamp_micros()))
+            .map(|ts| {
+                Some(ScalarRefImpl::Timestamptz(
+                    risingwave_common::types::Timestamptz::from_micros(ts),
+                ))
+            })
+    }
```

Comment:
Seems it is okay to simplify the return type to `DatumRef<'_>`, and then it can be:
```suggestion
    pub fn extract_timestamp(&self) -> DatumRef<'_> {
        Some(ScalarRefImpl::Timestamptz(
            self.timestamp?.to_chrono_utc().ok()?.into(),
        ))
    }
```